src/ButteredBread.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.25;
+
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {ERC20VotesUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {IButteredBread} from "src/interfaces/IButteredBread.sol";
+
+/**
+ * @title Breadchain Buttered Bread
+ * @notice Deposit Butter (LP tokens) to earn scaling rewards
+ * @author Breadchain Collective
+ * @custom:coauthor @RonTuretzky
+ * @custom:coauthor @daopunk
+ */
+contract ButteredBread is ERC20VotesUpgradeable, OwnableUpgradeable, IButteredBread {
+    /// @notice Access control for Breadchain sanctioned liquidity pools
+    mapping(address lp => bool allow) public allowlistedLPs;
+    /// @notice How much ButteredBread should be minted for a Liquidity Pool token (Butter)
+    mapping(address lp => uint256 factor) public scalingFactors;
+    /// @notice Butter balance by account and Liquidity Pool token deposited
+    mapping(address account => mapping(address lp => uint256 balance)) public accountToLPBalances;
+
+    modifier isAllowed(address _lp) {
+        if (allowlistedLPs[_lp] != true) revert NotAllowListed();
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address[] memory _liquidityPools,
```

Comment:
believe these can all be "calldata" since they're in an initializer not a constructor